docs/examples/gen_velocities.rst

```diff
@@ -0,0 +1,432 @@
+:py:meth:`~astropy.coordinates.EarthLocation.get_gcrs` method, which returns
+positions (with respect to the centre of the Earth) as a function of time.
+These are transformed into the lens frame using the
+:py:meth:`~astropy.coordinates.BaseCoordinateFrame.transform_to` method.
+Velocities can then be extracted using the
+:py:attr:`~astropy.coordinates.BaseCoordinateFrame.velocity` attribute, and
+finally :py:attr:`~astropy.coordinates.CartesianDifferential.d_z` isolates the
+z-component of the velocity (in the direction of the screen).
+
+.. plot::
+    :include-source:
+    :context:
+
+    earth_loc = EarthLocation('148°15′47″E', '32°59′52″S')
+    
+    v_earth = earth_loc.get_gcrs(t).transform_to(lens_frame).velocity.d_z
+
+Pulsar's orbital velocity
+-------------------------
+
+Compute the pulsar's orbital velocity projected onto the screen
+    
+.. math::
+
+    v_\mathrm{p,orb} = - \frac{ K_\mathrm{p} }{ \sin( i_\mathrm{p} ) }
+                         \left[ \cos( i_\mathrm{p} )
+                                \sin( \mathit{\Delta\Omega}_\mathrm{p} )
+                                \cos( \theta_\mathrm{p} )
+                              - \cos( \mathit{\Delta\Omega}_\mathrm{p} )
+                                \sin( \theta_\mathrm{p} )
+                         \right].
+
+Here, :math:`\theta_\mathrm{p}( t )` is the phase of pulsar orbit as measured
+from its ascending node.
+
+.. plot::
+    :include-source:
+    :context:
+
+    th_p = ((t - t0_p) / p_b).to('') * u.cycle
+
+    v_p_orb = (-k_p / np.sin(i_p)
+               * (np.cos(i_p) * np.sin(delta_omega_p) * np.cos(th_p)
+                              - np.cos(delta_omega_p) * np.sin(th_p)))
+
+Pulsar system barycentre velocity
+---------------------------------
+
+The pulsar system barycentre velocity projected onto the screen is given by
+
+.. math::
+
+    v_\mathrm{p,pm} \simeq d_\mathrm{p}
+                           \left[ \mu_\alpha \sin( \mathit{\Omega}_\mathrm{s} )
+                                + \mu_\delta \cos( \mathit{\Omega}_\mathrm{s} )
+                           \right].
+
+This can be computed manually, but it can also be retrieved from the
+:py:class:`~astropy.coordinates.SkyCoord` of the pulsar system (which contains
+the system's proper motion) by transforming it to ``lens_frame``.
+
+.. plot::
+    :include-source:
+    :context:
+    
+    v_p_pm = psr_coord.transform_to(lens_frame).velocity.d_z
+
+Effective velocity
+------------------
+
+Combine the velocities of the pulsar, Earth, and the lens into the effective
+velocity
+
+.. math::
+
+    v_\mathrm{eff} = \frac{1}{s} v_\mathrm{ISM}
+                     - \frac{1 - s}{s} ( v_\mathrm{p,orb} + v_\mathrm{p,pm} )
+                     - v_\mathrm{E}
+
+.. plot::
+    :include-source:
+    :context:
+    
+    v_eff = 1. / s * v_ism - ((1 - s) / s) * (v_p_orb + v_p_pm) - v_earth
+
+
+Curvature and scaled effective velocity
```

Comment:
My feeling is that at this point one should first show those velocities and describe how they make sense. Then, the transition to curvature will make more sense.

Reply:
I added a figure showing the terms that make up v_eff, and v_eff itself. Is that what you had in mind, or rather the velocities themselves without the 1/s, (1-s)/s factors and signs.

I didn't add much of a description. I found the figure relatively self-explanatory, but we could still add a bit of text if you think it would be helpful.

Reply:
I did have in mind the actual velocities, so one can see that, e.g., that of the Earth is of the right order of magnitude, but the different components also makes a lot of sense. Let's stick with just that for now!